src/frolladaptive.c

```diff
@@ -1,32 +1,53 @@
 #include "data.table.h"
 
-/* fast adaptive rolling mean - router
- * algo = 0: fadaptiverollmeanFast
- *   first pass cumsum based solution, second pass uses cumsum to calculate answer
- * algo = 1: fadaptiverollmeanExact
- *   recalculate whole mean for each observation, roundoff correction is adjusted, also support for NaN and Inf
+/* rolling adaptive fun - router for fun and algo
+ * rfun enum rollfun_t routes to rolling function
+ * algo = 0: fast
+ *   adding/removing in/out of sliding window of observations
+ * algo = 1: exact
+ *   recalculate whole fun for each observation, for mean roundoff correction is adjusted, also support for NaN and Inf
  */
-void fadaptiverollmean(unsigned int algo, double *x, uint64_t nx, ans_t *ans, int *k, double fill, bool narm, int hasna, bool verbose) {
+void frolladaptivefun(rollfun_t rfun, unsigned int algo, double *x, uint64_t nx, ans_t *ans, int *k, double fill, bool narm, int hasna, bool verbose) {
   double tic = 0;
   if (verbose)
     tic = omp_get_wtime();
-  if (algo==0) {
-    fadaptiverollmeanFast(x, nx, ans, k, fill, narm, hasna, verbose);
-  } else if (algo==1) {
-    fadaptiverollmeanExact(x, nx, ans, k, fill, narm, hasna, verbose);
+  switch (rfun) {
+  case MEAN :
+    if (algo==0) {
+      frolladaptivemeanFast(x, nx, ans, k, fill, narm, hasna, verbose);
+    } else if (algo==1) {
+      frolladaptivemeanExact(x, nx, ans, k, fill, narm, hasna, verbose);
+    }
+    break;
+  case SUM :
+    if (algo==0) {
+      frolladaptivesumFast(x, nx, ans, k, fill, narm, hasna, verbose);
+    } else if (algo==1) {
+      frolladaptivesumExact(x, nx, ans, k, fill, narm, hasna, verbose);
+    }
+    break;
+  case MAX :
+    if (algo==0 && verbose) {
+      //frolladaptivemaxFast(x, nx, ans, k, fill, narm, hasna, verbose); // frolladaptivemaxFast does not exists as of now
+      snprintf(end(ans->message[0]), 500, _("%s: algo %u not implemented, fall back to %u\n"), __func__, algo, (unsigned int) 1);
```

Comment:
algo 0 and algo 1 are less explanatory than algo fast and algo exact.
Especially for the case of frolladaptivemax users might be surprised what 0 and 1 mean.

Reply:
Yes, it was like this for simplicity rather than being informative. Rolling functions don't have verbose argument for that reason, as those messages are more technical ones. Reason to introduce them was ability to test which variant is running.

Reply:
If we want to change we could do it in separate PR as it is not related to changes in this PRs

Reply:
marked for follow-up: #5902